src/v/cluster/types.h

```diff
@@ -2795,6 +2795,41 @@ struct bootstrap_cluster_cmd_data
     std::vector<model::broker> initial_nodes;
 };
 
+enum class recovery_stage : int8_t {
+    // A recovery has been initialized. We've already downloaded and serialized
+    // the manifest. While in this state, a recovery manager may validate that
+    // the recovery materials are downloadable.
+    initialized = 0,
+
+    // Recovery steps are beginning. We've already validated that the recovery
+    // materials are downloadable, though these aren't persisted in the
+    // controller beyond the manifest (it is expected that upon leadership
+    // changes, they are redownloaded).
+    starting = 1,
+
+    recovered_license = 2,
+    recovered_cluster_config = 3,
+    recovered_users = 4,
+    recovered_acls = 5,
+    recovered_remote_topic_data = 6,
+    recovered_topic_data = 7,
+
+    // All state from the controller snapshot has been recoevered.
+    // Reconciliation attempts do not need to redownload the controller
+    // snapshot to proceed.
+    recovered_controller_snapshot = 8,
+
+    recovered_offsets_topic = 9,
+    recovered_tx_coordinator = 10,
+
+    // Recovery has completed successfully. This is a terminal state.
+    complete = 11,
+
+    // Recovery has failed. This is a terminal state.
+    failed = 12,
```

Comment:
1. are the numbers here going to be serialized, is that why they have explicit values?
2. is complete/failed intended to be at the end of the list? if so, should they be like 200 and 201 if we'd add more stages later or does the values not really matter?

Reply:
1. that's right
2. I don't feel strongly about it either way, done